text/0000-reactive-assignments.md

```diff
@@ -650,7 +650,14 @@ We can do better, **but it requires something potentially controversial** — a
 
 ### Server-side rendering
 
-TODO
+The compiler running with the `generate: 'ssr'` option produces completely different code from `generate: 'dom'`. This is because server-rendered components do not have a lifespan — they are rendered, then immediately discarded, so rather than having a concept of 'instances' an SSR component exports a function that given some data creates HTML and CSS.
+
+This does create some subtle behaviour differences however: there is no real place to do any kind of setup work, and `oncreate` (and the other lifecycle hooks) never run. Component bindings are also brittle.
```

Comment:
I think it's important to clarify here – if `oncreate` is to be renamed to `onmount` (aka, the component is attached to DOM now), then yes, it has no business being invoked in SSR context.

Only mention it because we briefly mentioned renamed `oncreate` to `onmount` (to serve as a clearer equivalent to `componentDidMount`). Then, perhaps, a _new_ `oncreate` would be added, serving as the `componentWillMount` equivalent – would _would_ make it redundant & useless in this RFC.

Reply:
There is arguably room for a `componentWillMount` equivalent, since there's currently nothing that runs a single time between the first time props are injected and the first time the component renders. It *is* achievable in userland using the `onprops` (are we calling it that?) function though.

👍 to `onmount`